src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,72 @@
-export const Pagination = () => {};
+import React from 'react';
+import classNames from 'classnames';
+import {
+  getLinkDataCy,
+  getLinkHref,
+  getLinkTextContent,
+} from './pagination-utils';
+
+export interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+  const listOfPages = [...Array(pagesAmount + 2)].map((_n, i) => i + 1);
+
+  function getOnClickCallback(index: number) {
+    switch (index) {
+      case 0:
+        return currentPage !== 1
+          ? () => onPageChange(currentPage - 1)
+          : () => {};
+
+      case pagesAmount + 1:
+        return currentPage !== pagesAmount
+          ? () => onPageChange(currentPage + 1)
+          : () => {};
+
+      default:
+        return () => onPageChange(index);
+    }
+  }
```

Comment:
The 'getOnClickCallback' function creates a new function each time it is called, which can lead to performance issues if not handled properly. It would be more efficient to define separate handlers for 'previous', 'next', and 'page number' clicks.